Makefile

```
@@ -65,36 +67,38 @@ docs-serve: ### Serve local docs
 gen-metastore: ## Run Metastore Code generation
 	@thrift -r --gen go --gen go:package_prefix=github.com/treeverse/lakefs/metastore/hive/gen-go/ -o metastore/hive metastore/hive/hive_metastore.thrift
 
-$(GOBINPATH)/swagger:
-	go get github.com/go-swagger/go-swagger/cmd/swagger
+go-mod-download: ## Download module dependencies
+	$(GOCMD) mod download
 
-gen-api: $(GOBINPATH)/swagger del-gen-api ## Run the go-swagger code generator
+go-install: go-mod-download ## Install dependencies
+	$(GOCMD) install github.com/go-swagger/go-swagger/cmd/swagger
+	$(GOCMD) install github.com/golang/mock/mockgen
+	$(GOCMD) install github.com/golangci/golangci-lint/cmd/golangci-lint
+	$(GOCMD) install github.com/rakyll/statik
+	$(GOCMD) install google.golang.org/protobuf/cmd/protoc-gen-go
+
+
+gen-api: go-install del-gen-api ## Run the go-swagger code generator
 	$(GOGENERATE) ./api/...
 
 del-gen-api:
 	@rm -rf $(API_BUILD_DIR)
 	@mkdir -p $(API_BUILD_DIR)
 
 .PHONY: gen-mockgen
-gen-mockgen: $(GOBINPATH)/mockgen ## Run the generator for inline commands
+gen-mockgen: go-install ## Run the generator for inline commands
 	$(GOGENERATE) ./graveler/committed/...
 	$(GOGENERATE) ./pyramid
 
-validate-swagger: $(GOBINPATH)/swagger  ## Validate swagger.yaml
+swagger-validator: go-install ## Validate swagger.yaml
```

```suggestion
swagger-validate: go-install
```
Most of our actions are verbs in the imperative mode.  And the comment adds little.

---

FYI the comment is parsed and shown as part of `make help`.. not sure that's a compelling enough reason to add it though

---

np, I'll replace all the XXX-validator to validate-XXX